pool/chainstate.go

```diff
@@ -235,41 +177,6 @@ func (cs *ChainState) prunePayments(ctx context.Context, height uint32) error {
 	return nil
 }
 
-// pruneJobs removes all jobs with heights less than the provided height.
-func (cs *ChainState) pruneJobs(height uint32) error {
```

Comment:
I think the chainstate should maintain the pruning functions like you did for `pruneAcceptedWork` and extracting a helper fetching function out of that instead.